paymentsheet/src/main/java/com/stripe/android/link/theme/Color.kt

```diff
@@ -72,62 +59,104 @@ internal object LinkThemeConfig {
+        iconTertiary = Neutral500,
+        iconWhite = Neutral0,
+        iconBrand = Brand200,
+        iconCritical = Critical500
     )
 
-    private val colorsDark = colorsLight.copy(
-        primary = LinkTeal,
-        secondary = DarkFill,
-        background = DarkBackground,
-        textPrimary = DarkTextPrimary,
-        textSecondary = DarkTextSecondary,
-        componentBackground = DarkComponentBackground,
-        componentBorder = DarkComponentBorder,
-        componentDivider = DarkComponentDivider,
-        progressIndicator = DarkProgressIndicator,
-        linkLogo = DarkLinkLogo,
-        closeButton = DarkCloseButton,
-        disabledText = DarkTextDisabled,
-        secondaryButtonLabel = DarkSecondaryButtonLabel,
-        otpElementColors = OTPElementColors(
-            selectedBorder = LinkTeal,
-            placeholder = DarkOtpPlaceholder
-        ),
+    private val colorsDark = LinkColors(
+        surfacePrimary = Neutral900,
+        surfaceSecondary = Neutral800,
+        surfaceTertiary = Neutral700,
+        surfaceBackdrop = Neutral900,
+        borderDefault = Neutral900,
+        borderSelected = Brand200,
+        borderCritical = Critical500,
+        buttonPrimary = Neutral200,
+        buttonSecondary = Neutral700,
+        buttonTertiary = Neutral800,
+        buttonBrand = Brand200,
+        buttonCritical = Critical600,
+        textPrimary = Neutral0,
+        textSecondary = Neutral300,
+        textTertiary = Neutral500,
+        textWhite = Neutral0,
+        textBrand = Brand200,
+        textCritical = Critical600,
+        iconPrimary = Neutral100,
+        iconSecondary = Neutral500,
+        iconTertiary = Neutral500,
+        iconWhite = Neutral0,
+        iconBrand = Brand200,
+        iconCritical = Critical500
     )
+
+    /**
+     * Workaround:
+     *
+     * - The new Link theme primary button uses white on dark mode and dark on light mode
+     * - But we're still using Link green theming for buttons, regardless of dark mode
+     * - This means that the fixed button color is not consistent with variable text / divider colors,
+     *   so we need to keep them fixed until we migrate to the updated primary color styling.
+     */
+    internal val LinkColors.contentOverPrimaryButton
+        get() = Neutral900
+    internal val LinkColors.separatorOverPrimaryButton
+        get() = Brand400
+
+    /**
+     * Workaround:
+     *
+     * Border color doesn't look great for radio buttons on dark mode. We give it a clearer
+     * color here.
+     *
+     */
+    internal val LinkColors.radioButtonColors
+        @Composable
+        get() = RadioButtonDefaults.colors(
+            selectedColor = LinkTheme.colors.buttonBrand,
+            unselectedColor = if (isSystemInDarkTheme()) Neutral700 else LinkTheme.colors.borderDefault
+        )
 }
 
 @Composable
 internal fun StripeThemeForLink(
+    materialPrimary: Color = LinkTheme.colors.textPrimary,
```

Comment:
For certain material-based Stripe components, we want to override the primary color to maintain consistency and match designs.